src/main/java/customerproductorder/web/CustomerEditServlet.java

```
@@ -39,10 +45,12 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp)
 
     private void saveCustomer(HttpServletRequest req) {
         HttpSession session = req.getSession(true);
-        Customer customer = (Customer) session.getAttribute("editCustomer");
-        int index = dataContainer.getCustomers().indexOf(customer);
-        dataContainer.getCustomers().set(index, new Customer(req.getParameter("firstName"),
-                req.getParameter("lastName"), req.getParameter("address"),
-                customer.getCardNumber()));
+        int id = (Integer)session.getAttribute("editCustomerId");
+        try{
+        service.save(new Customer(req.getParameter("firstName"),
+                req.getParameter("lastName"), req.getParameter("address"),id));}
+        catch(DaoException e){
```

One more time unhandled and swallowed exception... Shame on you...